### ISSUE TITLE
Optional pair

### ISSUE THREAD

**Issue by HGuillemet** [bytedeco/javacpp#718]
The JNI of container classes generated by:
```
infoMap.put(new Info("std::pair<int,int>").pointerTypes("IntPair").define());
infoMap.put(new Info("std::optional<std::pair<int,int> >").pointerTypes("IntPairOptional").define());
```
does't compile, with errors like:
```
error: 'class std::optional<std::pair<int, int> >' has no member named 'first'
 1452 |     int rval = ptr->first;
```
The `optional` container passes the `first` and `second` function members of `std::pair` through, which can be practical, but there seems to be missing an indirection. The line above should be:
```C++
int rval = (*ptr)->first;
```

**Comment by saudet** [comment]
Fixed in commit 78f0203d3a8ce1b4f410ab52c1fd4bbda0e500d5

**Comment by HGuillemet** [comment]
Thanks.
